### ISSUE TITLE
Error on some failed tests

### ISSUE THREAD

**Issue by panrafal** [litixsoft/karma-mocha-reporter#40]
Sometimes, when printing details about failed tests reporter throws below error and causes browser to permanently disconnect.
I say sometime, because on most of the failed tests the details are printed correctly, including the expected/actual diff.

```
Missing error handler on `socket`.
TypeError: Cannot read property 'expected' of undefined
    at printFailures (/opt/node_modules/karma-mocha-reporter/index.js:309:63)
    at printFailures (/opt/node_modules/karma-mocha-reporter/index.js:345:17)
    at printFailures (/opt/node_modules/karma-mocha-reporter/index.js:345:17)
    at printFailures (/opt/node_modules/karma-mocha-reporter/index.js:345:17)
    at [object Object].self.onRunComplete (/opt/node_modules/karma-mocha-reporter/index.js:543:21)
    at [object Object].<anonymous> (/opt/node_modules/karma/lib/events.js:13:22)
    at emitTwo (events.js:92:20)
    at [object Object].emit (events.js:172:7)
    at [object Object].onBrowserComplete (/opt/node_modules/karma/lib/executor.js:47:15)
    at [object Object].<anonymous> (/opt/node_modules/karma/lib/events.js:13:22)
    at emitTwo (events.js:92:20)
    at [object Object].emit (events.js:172:7)
    at [object Object].onComplete (/opt/node_modules/karma/lib/browser.js:142:13)
    at Socket.<anonymous> (/opt/node_modules/karma/lib/events.js:13:22)
    at emitTwo (events.js:92:20)
```

**Comment by 4kochi** [comment]
Seems like the property `assertionErrors` is not set in some cases by `karma-mocha`. Can you please send me a test case which leads to that error. Maybe i can find something.

Anyway i will add an additional check if the property exists to prevent the exception.

**Comment by straku** [comment]
I am encountering the same error but only when `showDiff` option is set to `true`.

**Comment by 4kochi** [comment]
Released as v1.2.1. Please check if it works now, otherwise if have to add more checks.

**Comment by straku** [comment]
Works like a charm! Thank you :smile:
